DataFormats/Detectors/TRD/src/TrapConfig3.cxx

```diff
@@ -0,0 +1,818 @@
+{
+  // there is no index for this but its not used online
+  for (auto& reg : mTrapRegisters) {
+    if (reg.getName() == name) {
+      return reg.getAddr();
+    }
+  }
+  return -1; // error condition
+}
+
+o2::trd::TrapRegInfo* TrapConfig3::getTrapRegInfoByAddr(uint32_t addr)
+{
+  int regidx = mTrapRegistersAddressIndexMap[addr];
+  if ((regidx < 0) && (regidx > kLastReg)) {
+    return nullptr; //&mTrapRegisters[0];
+  }
+  return &mTrapRegisters[regidx];
+}
+
+o2::trd::TrapRegInfo* TrapConfig3::getTrapRegInfoByIdx(uint32_t idx)
+{
+  if ((idx < 0) && (idx > kLastReg)) {
+    return nullptr;
+  }
+  return &mTrapRegisters[idx];
+}
+
+void TrapConfig3::getRegisterByAddr(uint32_t registeraddr, std::string& regname, int32_t& newregidx, int32_t& numberbits)
+{
+  int idx = -1;
+  idx = getRegIndexByAddr(registeraddr);
+  if (idx >= 0) {
+    regname = mTrapRegisters[idx].getName();
+    numberbits = mTrapRegisters[idx].getNbits();
+    newregidx = idx;
+  } else {
+    regname = "";
+    numberbits = -1;
+    newregidx = -1;
+  }
+}
+
+// get all the values for a given register values for a given mcm
+void TrapConfig3::getAllRegisters(int mcmidx, std::array<uint32_t, kLastReg>& mcmregisters)
+{
+  for (int reg = 0; reg < kLastReg; ++reg) {
+    mcmregisters[reg] = getRegisterValueByIdx(reg, mcmidx);
+  }
+}
+
+// get all the values for a given register, all mcms
+void TrapConfig3::getAllMCMByIndex(int regidx, std::array<uint32_t, o2::trd::constants::MAXMCMCOUNT>& mcms)
+{
+  if ((regidx >= 0) && (regidx < kLastReg)) {
+  }
+  for (int mcm = 0; mcm < o2::trd::constants::MAXMCMCOUNT; ++mcm) {
+    mcms[mcm] = getRegisterValueByIdx(regidx, mcm);
+  }
+}
+
+// get all the values for a given register name, all mcms
+void TrapConfig3::getAllMCMByName(std::string registername, std::array<uint32_t, o2::trd::constants::MAXMCMCOUNT>& mcms) // return all the mcm values for a particular register name
+{
+  int regindex = getRegIndexByName(registername);
+  for (int mcm = 0; mcm < o2::trd::constants::MAXMCMCOUNT; ++mcm) {
+    mcms[mcm] = getRegisterValueByIdx(regindex, mcm);
+  }
+}
+
+// get all the values for a given register address, all mcms
+void TrapConfig3::getAllMCMByAddress(int registeraddress, std::array<uint32_t, o2::trd::constants::MAXMCMCOUNT>& mcms)
+{
+  int regindex = getRegIndexByAddr(registeraddress);
+  for (int mcm = 0; mcm < o2::trd::constants::MAXMCMCOUNT; ++mcm) {
+    mcms[mcm] = getRegisterValueByIdx(regindex, mcm);
+  }
+}
+
+// get all the values for a given register values for all the mcm
+void TrapConfig3::getAll(std::array<uint32_t, kLastReg * o2::trd::constants::MAXMCMCOUNT>& configdata)
+{
+  for (int mcm = 0; mcm < constants::MAXMCMCOUNT; ++mcm) {
+    for (int reg = 0; reg < kLastReg; ++reg) {
+      configdata[mcm * kLastReg + reg] = getRegisterValueByIdx(reg, mcm);
+    }
+  }
+}
+
+bool TrapConfig3::printRegister(TrapRegInfo* reg, int det, int rob, int mcm)
```

Comment:
This doesn't seem to do anything?

Reply:
it was used for debugging purposes only.